pilot/pkg/bootstrap/server.go

```diff
@@ -1281,3 +1287,60 @@ func (s *Server) initStatusManager(_ *PilotArgs) {
 		return nil
 	})
 }
+
+// initLocalSecretListener creates a new secret listener to listen for
+// secrets which represent new multi cluster participating clusters in a mesh.
+func (s *Server) initLocalSecretListener(args *PilotArgs) {
+	// Check if the istiod's kube client API endpoint is the same as the local cluster API endpoint
+	// If they are the same, then the Istiod is in cluster Istiod, otherwise, it will be an
+	// external istiod running outside of the config cluster, the localKubeClient and the
+	// istiod kube client endpoint should have been different.
+	config, err := rest.InClusterConfig()
```

Comment:
can you draw a diagram to show your solution?


What is `s.kubeClient` in istiod less?